src/Jenssegers/Mongodb/Relations/BelongsTo.php

```diff
@@ -69,4 +72,29 @@ protected function whereInMethod(EloquentModel $model, $key)
     {
         return 'whereIn';
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function match(array $models, Collection $results, $relation)
```

Comment:
I suggest i we using type hints, add type for all arguments. Maybe for this we require to add php: "^7.1" to composer.json for this case

Reply:
Probably yes and no because all illuminate/support and etc already requires php ^7.1.3